test/common/grpc/grpc_web_filter_test.cc

```diff
@@ -29,73 +31,60 @@ const char TRAILERS[] = "\x80\x00\x00\x00\x20grpc-status:0\r\ngrpc-message:ok\r\
-  EXPECT_EQ(std::string(MESSAGE, MESSAGE_SIZE), TestUtility::bufferToString(request_buffer));
+  bool isTextRequest() const {
+    return request_content_type() == Http::Headers::get().ContentTypeValues.GrpcWebText ||
+           request_content_type() == Http::Headers::get().ContentTypeValues.GrpcWebTextProto;
+  }
 
-  // Tests request trailers, they are passed through.
-  Http::TestHeaderMapImpl request_trailers;
-  request_trailers.addViaCopy(Http::Headers::get().GrpcStatus, "0");
-  request_trailers.addViaCopy(Http::Headers::get().GrpcMessage, "ok");
-  EXPECT_EQ(Http::FilterTrailersStatus::Continue, filter_.decodeTrailers(request_trailers));
-  EXPECT_STREQ("0", request_trailers.GrpcStatus()->value().c_str());
-  EXPECT_STREQ("ok", request_trailers.GrpcMessage()->value().c_str());
+  bool isBinaryRequest() const {
+    return request_content_type() == Http::Headers::get().ContentTypeValues.GrpcWeb ||
+           request_content_type() == Http::Headers::get().ContentTypeValues.GrpcWebProto;
+  }
 
-  // Tests response headers.
-  Http::TestHeaderMapImpl response_headers;
-  response_headers.addViaCopy(Http::Headers::get().Status, "200");
-  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_.encodeHeaders(response_headers, false));
-  EXPECT_EQ("200", response_headers.get_(Http::Headers::get().Status.get()));
-  EXPECT_EQ(Http::Headers::get().ContentTypeValues.GrpcWeb,
-            response_headers.ContentType()->value().c_str());
+  bool accept_text_response() {
+    return request_accept() == Http::Headers::get().ContentTypeValues.GrpcWebText ||
+           request_accept() == Http::Headers::get().ContentTypeValues.GrpcWebTextProto;
+  }
 
-  // Tests response data.
-  Buffer::OwnedImpl response_buffer(MESSAGE, MESSAGE_SIZE);
-  EXPECT_EQ(Http::FilterDataStatus::Continue, filter_.encodeData(response_buffer, false));
-  EXPECT_EQ(std::string(MESSAGE, MESSAGE_SIZE), TestUtility::bufferToString(response_buffer));
-  response_buffer.drain(response_buffer.length());
+  bool accept_binary_response() {
+    return request_accept() == Http::Headers::get().ContentTypeValues.GrpcWeb ||
+           request_accept() == Http::Headers::get().ContentTypeValues.GrpcWebProto;
+  }
 
-  // Tests response trailers.
-  Buffer::OwnedImpl trailers_buffer;
-  EXPECT_CALL(encoder_callbacks_, addEncodedData(_))
-      .WillOnce(Invoke([&](Buffer::Instance& data) { trailers_buffer.move(data); }));
-  Http::TestHeaderMapImpl response_trailers;
-  response_trailers.addViaCopy(Http::Headers::get().GrpcStatus, "0");
-  response_trailers.addViaCopy(Http::Headers::get().GrpcMessage, "ok");
-  EXPECT_EQ(Http::FilterTrailersStatus::Continue, filter_.encodeTrailers(response_trailers));
-  EXPECT_EQ(std::string(TRAILERS, TRAILERS_SIZE), TestUtility::bufferToString(trailers_buffer));
+  GrpcWebFilter filter_;
+  NiceMock<Http::MockStreamEncoderFilterCallbacks> encoder_callbacks_;
+};
+
+TEST_F(GrpcWebFilterTest, SupportedContentTypes) {
```

Comment:
It's a TEST_F, will be run only once.
Here's the test log. FYI.

-----------------------------------------------------------------------------
[==========] Running 17 tests from 2 test cases.
[----------] Global test environment set-up.
[----------] 1 test from GrpcWebFilterTest
[ RUN      ] GrpcWebFilterTest.SupportedContentTypes
[       OK ] GrpcWebFilterTest.SupportedContentTypes (1 ms)
[----------] 1 test from GrpcWebFilterTest (1 ms total)

[----------] 16 tests from Unary/GrpcWebFilterTest
[ RUN      ] Unary/GrpcWebFilterTest.Unary/0
[       OK ] Unary/GrpcWebFilterTest.Unary/0 (0 ms)
[ RUN      ] Unary/GrpcWebFilterTest.Unary/1
[       OK ] Unary/GrpcWebFilterTest.Unary/1 (1 ms)
[ RUN      ] Unary/GrpcWebFilterTest.Unary/2
[       OK ] Unary/GrpcWebFilterTest.Unary/2 (0 ms)
[ RUN      ] Unary/GrpcWebFilterTest.Unary/3
[       OK ] Unary/GrpcWebFilterTest.Unary/3 (0 ms)
[ RUN      ] Unary/GrpcWebFilterTest.Unary/4
[       OK ] Unary/GrpcWebFilterTest.Unary/4 (1 ms)
[ RUN      ] Unary/GrpcWebFilterTest.Unary/5
[       OK ] Unary/GrpcWebFilterTest.Unary/5 (0 ms)
[ RUN      ] Unary/GrpcWebFilterTest.Unary/6
[       OK ] Unary/GrpcWebFilterTest.Unary/6 (1 ms)
[ RUN      ] Unary/GrpcWebFilterTest.Unary/7
[       OK ] Unary/GrpcWebFilterTest.Unary/7 (0 ms)
[ RUN      ] Unary/GrpcWebFilterTest.Unary/8
[       OK ] Unary/GrpcWebFilterTest.Unary/8 (0 ms)
[ RUN      ] Unary/GrpcWebFilterTest.Unary/9
[       OK ] Unary/GrpcWebFilterTest.Unary/9 (1 ms)
[ RUN      ] Unary/GrpcWebFilterTest.Unary/10
[       OK ] Unary/GrpcWebFilterTest.Unary/10 (0 ms)
[ RUN      ] Unary/GrpcWebFilterTest.Unary/11
[       OK ] Unary/GrpcWebFilterTest.Unary/11 (0 ms)
[ RUN      ] Unary/GrpcWebFilterTest.Unary/12
[       OK ] Unary/GrpcWebFilterTest.Unary/12 (1 ms)
[ RUN      ] Unary/GrpcWebFilterTest.Unary/13
[       OK ] Unary/GrpcWebFilterTest.Unary/13 (0 ms)
[ RUN      ] Unary/GrpcWebFilterTest.Unary/14
[       OK ] Unary/GrpcWebFilterTest.Unary/14 (1 ms)
[ RUN      ] Unary/GrpcWebFilterTest.Unary/15
[       OK ] Unary/GrpcWebFilterTest.Unary/15 (0 ms)
[----------] 16 tests from Unary/GrpcWebFilterTest (6 ms total)

[----------] Global test environment tear-down
[==========] 17 tests from 2 test cases ran. (7 ms total)
[  PASSED  ] 17 tests.